src/app/containers/StakePage/components/LegacySovFeeBlock.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useCallback, useMemo, useEffect } from 'react';
+import { useCachedAssetPrice } from 'app/hooks/trading/useCachedAssetPrice';
+import { useAccount } from 'app/hooks/useAccount';
+import { useTranslation } from 'react-i18next';
+import { getContract } from 'utils/blockchain/contract-helpers';
+import { Tooltip } from '@blueprintjs/core';
+import { weiTo18, weiTo4 } from 'utils/blockchain/math-helpers';
+import { useStaking_getLegacyVestingFees } from 'app/hooks/staking/useStaking_getLegacyVestingFees';
+import { translations } from 'locales/i18n';
+import { LoadableValue } from 'app/components/LoadableValue';
+import { bignumber } from 'mathjs';
+import { Asset } from 'types';
+import { weiToUSD } from 'utils/display-text/format';
+import { useSendToContractAddressTx } from 'app/hooks/useSendToContractAddressTx';
+import { TransactionDialog } from 'app/components/TransactionDialog';
+import { Button, ButtonSize, ButtonStyle } from 'app/components/Button';
+import { IBaseFeeBlockProps } from './FeeBlock';
+import VestingABI from 'utils/blockchain/abi/Vesting.json';
+import { AbiItem } from 'web3-utils';
+
+interface ILegacySovFeeBlock extends IBaseFeeBlockProps {
+  vestingContract: string;
+}
+
+export const LegacySovFeeBlock: React.FC<ILegacySovFeeBlock> = ({
+  contractToken,
+  updateUsdTotal,
+  title,
+  frozen,
+  vestingContract,
+}) => {
+  const account = useAccount();
+  const { asset } = contractToken;
+  const { t } = useTranslation();
+  const token = useMemo(() => contractToken.getTokenContractName(), [
+    contractToken,
+  ]);
+  const dollars = useCachedAssetPrice(asset, Asset.USDT);
+  const tokenAddress = getContract(token)?.address;
+
+  const currency = useStaking_getLegacyVestingFees(
+    vestingContract,
+    tokenAddress,
+  );
+
+  const dollarValue = useMemo(() => {
+    if (currency.value === null) {
+      return '';
+    }
+    return bignumber(currency.value)
+      .mul(dollars.value)
+      .div(10 ** contractToken.decimals)
+      .toFixed(0);
+  }, [dollars.value, currency.value, contractToken.decimals]);
+
+  const { send, ...tx } = useSendToContractAddressTx(
+    vestingContract,
+    VestingABI as AbiItem[],
+    'collectDividends',
+  );
+  const handleWithdrawFee = useCallback(
+    async e => {
+      e.preventDefault();
+      try {
+        send([tokenAddress, 100, account]);
```

Comment:
Create a const `maxCheckpoints` and use it instead of `100`. Also, just to be sure, use `account.toLowerCase()`.